test/media/timestamp_offset_corrector_unit.js

```diff
@@ -0,0 +1,181 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+goog.require('shaka.media.TimestampOffsetCorrector');
+goog.require('shaka.media.SegmentReference');
+goog.require('shaka.media.InitSegmentReference');
+goog.require('shaka.util.ManifestParserUtils');
+
+describe('TimestampOffsetCorrector', () => {
+  const initSegmentUri =
+      '/base/test/test/assets/timestamp-offset-corrector-header.m4s';
+  const mediaSegmentUri =
+      '/base/test/test/assets/timestamp-offset-corrector-segment.m4s';
+
+  const contentType = shaka.util.ManifestParserUtils.ContentType.AUDIO;
+
+  /** @type {!ArrayBuffer} */
+  let initSegment;
+  /** @type {!ArrayBuffer} */
+  let mediaSegment;
+
+  // Actual base media decode time for the media segment
+  const baseMediaDecodeTimeSec = 16628739.423377778;
+
+  const initSegmentReference = new shaka.media.InitSegmentReference(
+      () => [initSegmentUri], // uri
+      0, // startByte
+      null); // endByte
+
+  const createSegmenRefWithTimestampOffset = (timestampOffset) => {
+    return new shaka.media.SegmentReference(
+        0, // startTime
+        2, // endTime
+        () => [mediaSegmentUri], // uris
+        0, // startByte
+        null, // endByte
+        initSegmentReference, // initSegmentReference
+        timestampOffset, // timestampOffset
+        0, // appendWindowStart
+        Infinity); // appendWindowEnd
+  };
+
+  const createStreamingConfig =
+      (correctTimestampOffset, maxTimestampDiscrepancy) => {
+        const config = shaka.util.PlayerConfiguration.createDefault().streaming;
+        config.correctTimestampOffset = correctTimestampOffset;
+        config.maxTimestampDiscrepancy = maxTimestampDiscrepancy;
+        return config;
+      };
+
+  beforeAll(async () => {
+    const responses = await Promise.all([
+      shaka.test.Util.fetch(initSegmentUri),
+      shaka.test.Util.fetch(mediaSegmentUri),
+    ]);
+    initSegment = responses[0];
+    mediaSegment = responses[1];
+  });
+
+  /** @type shaka.media.TimestampOffsetCorrector */
+  let tsoc;
+
+  let onEvent;
+
+  beforeEach(() => {
+    onEvent = jasmine.createSpy('onEvent');
+    tsoc = new shaka.media.TimestampOffsetCorrector(
+        shaka.test.Util.spyFunc(onEvent));
+  });
+
+  describe('checkTimestampOffset()', () => {
+    it('does not alter correct timestampOffset', () => {
+      tsoc.configure(createStreamingConfig(true, 10));
+      const segRef =
+          createSegmenRefWithTimestampOffset(-baseMediaDecodeTimeSec);
+      tsoc.parseTimescalesFromInitSegment(contentType, initSegment);
+      const corrected =
+          tsoc.checkTimestampOffset(contentType, segRef, mediaSegment);
+      expect(corrected === false);
```

Comment:
This isn't doing anything.  You need to write something like expect(corrected).toBe(false).

What you've written is equivalent to expect(false) or expect(true), but that's an incomplete statement from Jasmine's point of view.  It says "here's a value to test", but not what the expectation is.